File: databricks/notebooks/tools/scala_tools/DumpExperiment.scala
// Databricks notebook source
// MAGIC %run ./DumpExperiment.scala

// COMMAND ----------

dbutils.widgets.text(" Experiment ID", "")
dbutils.widgets.text("Artifact Max Level", "1")
dbutils.widgets.dropdown("Show run info","no",Seq("yes","no"))
dbutils.widgets.dropdown("Show run data","no",Seq("yes","no"))
dbutils.widgets.dropdown("In Databricks","no",Seq("yes","no"))

val experimentId = dbutils.widgets.get(" Experiment ID")
val artifactMaxLevel = dbutils.widgets.get("Artifact Max Level").toInt
val showInfo = dbutils.widgets.get("Show run info") == "yes"
val showData = dbutils.widgets.get("Show run data") == "yes"
val inDatabricks = dbutils.widgets.get("In Databricks") == "yes"

// COMMAND ----------

import org.mlflow.tracking.MlflowClient
DumpExperiment.dumpExperiment(new MlflowClient(), experimentId, artifactMaxLevel, showInfo, showData, inDatabricks)

// COMMAND ----------

import org.mlflow.tracking.MlflowClient
DumpExperiment.dumpExperiment(new MlflowClient(), experimentId, artifactMaxLevel, showInfo, showData, !inDatabricks)

// COMMAND ----------

dumpExperiment2(new MlflowClient(), experimentId, artifactMaxLevel, showInfo, showData)

File: databricks/notebooks/tools/dump_experiment.py
# Databricks notebook source
# MAGIC %md ### Dump experiment with all its run and artifact details
# MAGIC * Shows params, metrics and tags
# MAGIC * Recursively shows all artifacts up to the specified level
# MAGIC * Note: Makes lots of calls to API, so beware running with Default experiment 0

# COMMAND ----------

dbutils.widgets.text(" Experiment ID or name", "1")
dbutils.widgets.text("Artifact Max Level", "1")
dbutils.widgets.dropdown("Show runs","yes",["yes","no"])

experiment_id_or_name = dbutils.widgets.get(" Experiment ID or name")
artifact_max_level = int(dbutils.widgets.get("Artifact Max Level"))
show_runs = dbutils.widgets.get("Show runs") == "yes"

experiment_id_or_name, artifact_max_level, show_runs

# COMMAND ----------

# MAGIC %run ./dump_experiment.py

# COMMAND ----------

dump_experiment(experiment_id_or_name, artifact_max_level, show_runs)